### DISCUSSION TITLE
Adding Repository Dependencies

### DISCUSSION THREAD

**Discussion by TimUbbink** [cloudreach/sceptre#839] [Q&A]
Hi there,

I was wondering if there is a possibility to add a dependency of a repo in a stack config? In my case not all files are hard dependencies for example.

I have a repository 'A' which is ever changing, and I have a file in another repo 'B' that is dependend on the files in repo 'A'; some will be added or removed. Is there a possibility to add a dependency for the file in repo 'B' on all the (current) files in repo 'A'? I don't want to update the dependencies whenever we are changing something in repo 'A'.

So B/example.yaml will look like this:

...
dependencies:
 -  A/dependent.yaml (hard dependency)
 -  A/* (soft dependency)

Cheers,

Tim

**Comment by zaro0508** [comment]
Sure, anything is possible.  Contributions are welcome, just throw up a PR and someone will review :)

**Comment by ngfgrant** [comment]
So we don’t differentiate between hard and soft dependencies as something like a symlink would. Something either depends on something else or it does not. 

The question is more around wildcard dependencies. That is something we could probably support without too much trouble. 

https://github.com/Sceptre/sceptre/blob/master/sceptre/config/reader.py#L308

And 

https://github.com/Sceptre/sceptre/blob/master/sceptre/config/reader.py#L240


Would be places to look to start to make these changes.
